staging/src/k8s.io/component-base/logs/internal/setverbositylevel/setverbositylevel.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package setverbositylevel stores callbacks that will be invoked by logs.GlogLevel.
+//
+// This is a separate package to avoid a dependency from
+// k8s.io/component-base/logs (uses the callbacks) to
+// k8s.io/component-base/logs/api/v1 (adds them). Not all users of the logs
+// package also use the API.
+package setverbositylevel
```

Comment:
Interface of this package is a little too open and doesn't follow golang naming scheme (https://go.dev/doc/effective_go#package-names). 

Could we expose `AddCallback` and `SetVerbosity` functions instead of `Mutex` and `Callbacks` to avoid misuse?

Instead naming package `setverbositylevel` could we rename it to just `verbosity`? This would allow external packages to call function `verbosity.Set`.

Reply:
> doesn't follow golang naming scheme (https://go.dev/doc/effective_go#package-names).

`setverbositylevel.Mutex` and `setverbositylevel.Callbacks` sounds okay to me.

> to avoid misuse?

This is an internal package - nothing outside of k8s.io/component-base/logs has access to it. Therefore I wasn't worried about preventing misuse and instead went for simplicity. Does that make sense?

> Instead naming package setverbositylevel could we rename it to just verbosity?

I can, but is `verbosity.Callbacks` really a better name?

Reply:
I don't think it's worth blocking, however I still think `verbosity` is a better package name. 

> I can, but is verbosity.Callbacks really a better name?

Maybe `verbosity.ChangeCallbacks`. Usually you would prefer `verbosity.RegisterCallback`.

Reply:
Let's continue with the simpler approach of not adding functions to the package. The package only exists to avoid circular package dependencies, not to protect variables against undesired access.